services/compute/virtualmachine/virtualmachine.go

```diff
@@ -418,6 +425,59 @@ func (c *client) getWssdVirtualMachineGuestAgentConfiguration(s *compute.GuestAg
 	return gac, nil
 }
 
+func (c *client) getWssdVirtualMachineProxyConfiguration(proxyConfig *compute.ProxyConfiguration, opType wssdcloudproto.Operation) (*wssdcloudproto.ProxyConfiguration, error) {
+	if proxyConfig == nil {
+		return nil, nil
+	}
+
+	proxyConfiguration := &wssdcloudproto.ProxyConfiguration{}
+
+	if proxyConfig.TrustedCa != nil {
+		if opType == wssdcloudproto.Operation_POST {
+			err := auth.CertCheck([]byte(*proxyConfig.TrustedCa))
+			if err != nil {
+				return nil, errors.Wrapf(errors.InvalidInput, err.Error())
+			}
+		}
+		proxyConfiguration.TrustedCa = *proxyConfig.TrustedCa
+	}
+
+	if proxyConfig.HttpProxy != nil {
+		if opType == wssdcloudproto.Operation_POST {
+			parsedUrl, err := validations.ValidateProxyURL(*proxyConfig.HttpProxy)
+			if err != nil {
+				return nil, errors.Wrapf(errors.InvalidInput, err.Error())
+			}
+			err = validations.TestProxyUrlConnection(parsedUrl, *proxyConfig.TrustedCa, "")
```

Comment:
I think we should not have URL connection check from moc-sdk. we may not have line of sight.

Reply:
why?

Reply:
Moc-sdk can be called from any host, it need not have access to the proxy. Only the VM needs to have access, we should be validating from SDK